invokeai/frontend/web/src/features/canvas/hooks/useCanvasZoom.ts

```diff
@@ -30,40 +33,57 @@ const useCanvasWheel = (stageRef: MutableRefObject<Konva.Stage | null>) => {
 
       e.evt.preventDefault();
 
-      const cursorPos = stageRef.current.getPointerPosition();
+      let delta = e.evt.deltaY;
 
-      if (!cursorPos) {
-        return;
-      }
+      // checking for ctrl key is pressed or not, 
+      // so that brush size can be controlled using ctrl + scroll up/down
+      if ($ctrl.get()) {
+        let size = brushSize;
 
-      const mousePointTo = {
-        x: (cursorPos.x - stageRef.current.x()) / stageScale,
-        y: (cursorPos.y - stageRef.current.y()) / stageScale,
-      };
+        if (delta < 0) {
+          if (brushSize - 5 <= 5) {
+            size = Math.max(brushSize - 1, 1);
+          } else {
+            size = Math.max(brushSize - 5, 1);
+          }
+        } else {
+          size = Math.min(brushSize + 5, 500);
+        }
```

Comment:
The scaling here doesn't feel right. For larger brush sizes, changing by 5 is way too slow, and for smaller brush sizes, 1 is too much. Also, the change is a flat 1 or 5. 

`e.evt.deltaY` is proportional to the speed of the scroll gesture. For example, a slow trackpad scroll might give a deltaY around +/-1, while a fast scroll is +/-15 or more. This is valuable user input data, we shouldn't discard this information. So `e.evt.deltaY` should be used in the calculation.

But we can't add deltaY to the old brush size directly - it will still be too large small brush sizes. I'd suggest dividing `e.evt.deltaY` by a small number (maybe 3?), then exponentiating the result (maybe by 2?). Then add this final delta to the current brush size.

This will give much finer control at small deltaY values (slower trackpad scrolls), and coarser control at high deltaY values (faster trackpad scrolls).

There will still be a problem - if you scroll fast at small brush sizes, the brush size will still change too quickly. So the delta also needs a max, in addition to the scaling. The max delta should be related to the current brush size so that the size can be finely controlled even when the brush size is very small.

Suggest taking the max of some small fraction of the old brush size (maybe 5% ish?) and a reasonable max delta (maybe around 20?).

Overall, this will help the brush size respond very smoothy and intuitively for trackpad users.

Should also be tested with a mouse scroll wheel.

Reply:
I looked into the code for changing the brush size normally and copied from there :)